### ISSUE TITLE
google_storage_object_acl googleapi: Error 403: The owner of the resource is required to have OWNER access., forbidden

### ISSUE THREAD

**Issue by sebglon** [terraform-providers/terraform-provider-google#1662]
### Terraform Version
Terraform v0.11.7
+ provider.aws v1.23.0
+ provider.google v1.14.0
+ provider.local v1.1.0
+ provider.mailgun v0.1.0
+ provider.random v1.3.1
+ provider.template v1.0.0

### Affected Resource(s)
google_storage_object_acl

### Terraform Configuration Files
```hcl
resource "google_storage_object_acl" "store-acl" {
  bucket = "my-bucket"
  object = "mypath/myfile.txt"

  role_entity = [

    "READER:user-${google_service_account.deployer.email}",
  ]

  depends_on = [
    "google_service_account.deployer",
  ]
}
```
Not applyed with this config

```hcl
resource "google_storage_object_acl" "store-acl" {
  bucket = "my-bucket"
  object = "mypath/myfile.txt"

  role_entity = [
    "OWNER:project-owners-PROJECTUID",
    "OWNER:project-editors-PROJECTUID",
    "READER:project-viewers-PROJECTUID",
    "OWNER:user-sebastien.glon@akeneo.com",
    "READER:user-${google_service_account.deployer.email}",
  ]

  depends_on = [
    "google_service_account.deployer",
  ]
}
```

And withe the error message with this config

### Debug Output
* google_storage_object_acl.store_acl: Error updating ACL for object my-bucket: googleapi: Error 403: The owner of the resource is required to have OWNER access., forbidden

### Steps to Reproduce
1. `terraform apply`

**Comment by danawillow** [comment]
Hey @sebglon, looks like the formatting of your issue report got a bit wonky- mind fixing it? It would also be great to know what it is you were trying to do (I'm guessing change from the first config to the second?) and to get some debug logs (see https://www.terraform.io/docs/internals/debugging.html)

**Comment by DRrawlins** [comment]
Whatever the bug is, I am also seeing this in 1.14, I had another workstation on 1.12 that it was working on. as soon as I nuked the old plugin and it brought down 1.14 starting seeing the same behavior. When reverted back to 1.12(you guys need way better management, way to hard to do this for so many different repos) everything worked and went back to normal.

**Comment by danawillow** [comment]
@DRrawlins, do you have debug logs you'd be willing to share?

**Comment by DRrawlins** [comment]
@danawillow Sorry! I did, been super busy. The new plugin version 1.15 seems to have fixed this bug anyway.

**Comment by sebglon** [comment]
Always present for me on 1.15

**Comment by danawillow** [comment]
@sebglon, do you have debug logs you'd be willing to share? (https://www.terraform.io/docs/internals/debugging.html is what we're talking about when we say debug logs, by the way)

**Comment by rileykarson** [comment]
I believe this was fixed as part of either `1.20.0` or `2.0.0`. Feel free to reopen / file a new issue if that's not the case though.
